### ISSUE TITLE
Sometimes `zos create` works, sometimes it doesn't.

### ISSUE THREAD

**Issue by hickscorp** [zeppelinos/zos#930]
We've been stuck on that problem for a while now. When running `zos create Note --init initialize --args 64`, we sometimes get an error `Cannot read property 'absolutePath' of undefined`.
What I mean by "sometimes" is on certain projects, not on others.

So using the `-v` flag we were able to see a stack trace of the error. It looks like this:

```
[Error] TypeError: Cannot read property 'absolutePath' of undefined
    at BuildArtifacts.getSourcePathFromArtifact (/home/doodloo/.config/yarn/global/node_modules/zos-lib/lib/artifacts/BuildArtifacts.js:35:29)
    at BuildArtifacts.artifactsPaths.forEach (/home/doodloo/.config/yarn/global/node_modules/zos-lib/lib/artifacts/BuildArtifacts.js:21:37)
    at Array.forEach (<anonymous>)
    at new BuildArtifacts (/home/doodloo/.config/yarn/global/node_modules/zos-lib/lib/artifacts/BuildArtifacts.js:19:24)
    at Object.getBuildArtifacts (/home/doodloo/.config/yarn/global/node_modules/zos-lib/lib/artifacts/BuildArtifacts.js:12:12)
    at new ContractAST (/home/doodloo/.config/yarn/global/node_modules/zos-lib/lib/utils/ContractAST.js:16:56)
    at Object.contractMethodsFromAbi (/home/doodloo/.config/yarn/global/node_modules/zos-lib/lib/artifacts/Contract.js:77:25)
    at contractMethods (/home/doodloo/.config/yarn/global/node_modules/zos/lib/prompts/prompt.js:158:22)
    at Object.methodsList (/home/doodloo/.config/yarn/global/node_modules/zos/lib/prompts/prompt.js:130:12)
    at getCommandProps (/home/doodloo/.config/yarn/global/node_modules/zos/lib/commands/create.js:79:38)
    at Object.<anonymous> (/home/doodloo/.config/yarn/global/node_modules/zos/lib/prompts/init-params.js:25:29)
    at Generator.next (<anonymous>)
    at /home/doodloo/.config/yarn/global/node_modules/zos/lib/prompts/init-params.js:7:71
    at new Promise (<anonymous>)
    at __awaiter (/home/doodloo/.config/yarn/global/node_modules/zos/lib/prompts/init-params.js:3:12)
    at Object.promptForMethodParams [as default] (/home/doodloo/.config/yarn/global/node_modules/zos/lib/prompts/init-params.js:21:12)
    at Command.<anonymous> (/home/doodloo/.config/yarn/global/node_modules/zos/lib/commands/create.js:51:55)
    at Generator.next (<anonymous>)
    at fulfilled (/home/doodloo/.config/yarn/global/node_modules/zos/lib/commands/create.js:4:58)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```

So we tried adding some logs in the `BuildArtifacts` constructor - something like `console.log(artifactsPaths)`. To our surprise, we saw two inputs. The first one succeeds and looks like this:

```js
[ '/home/doodloo/project/build/contracts/Initializable.json',
  '/home/doodloo/project/build/contracts/Migrations.json',
  '/home/doodloo/project/build/contracts/Note.json' ]
```
This is absolutelly fine. However, just before the crash, another log of the same function shows:
```js
[ '/home/doodloo/project/package.json',
  '/home/doodloo/project/zos.dev-5777.json',
  '/home/doodloo/project/zos.json' ]
```

I'm not sure why `zos` is trying the top-level json files but that's obviously making it trip... Has anyone encountered this error before?

**Comment by spalladino** [comment]
Hey @hickscorp! Thanks for the report! Which version are you in? Could it be related to [this issue](https://github.com/zeppelinos/zos/issues/927)?

**Comment by jcarpanelli** [comment]
Hello, @hickscorp! We have fixed this bug yesterday in https://github.com/zeppelinos/zos/pull/928. We will release a patch version soon! Sorry for the inconvenience :see_no_evil:.

**Comment by hickscorp** [comment]
@spalladino @jcarpanelli :man_facepalming: sorry I missed this patch.
What version should I use until it's published?

**Comment by spalladino** [comment]
@hickscorp we've just released 2.3.1 with a fix. Mind giving it a shot?
